### PR TITLE
Merge brainio_contrib

### PR DESCRIPTION
**Pull request by stothe2**
- Merge code from brainio_contrib
- Move generic code from brainio_collection to brainio_base (see [PR](https://github.com/brain-score/brainio_base/pull/17))
- Added some checks for stimuli that is packaged for running on monkey machine (this is still work-in-progress, and will most likely be updated based on the type of stimuli folks are sharing)

This seemed too easy, so I'm guessing I've missed something. Things I can think off the top of my head that need to be checked:
- [ ] ~~Travis file OK? I've left it as it was originally in brainio_collection.~~
- [x] Readme: anything to add?
- [x] [Environment YML](https://github.com/brain-score/brainio_contrib/blob/master/environment.yml) file from brainio_contrib: I haven't added that yet. Should I be update [setup.py](https://github.com/brain-score/brainio_collection/blob/master/setup.py) instead?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,3 +4,86 @@
 
 Utility to retrieve and write to the BrainIO collection of stimulus sets and assemblies.
 
+
+# BrainIO contrib
```
Comment (mschrimpf) [inline README.md:8]: rename to "Packaging Instructions"? Maybe also fix README structure, the sentence above makes it a little weird -- could have a separate "BrainIO collection" header at the very top with some description underneath

File: README.md [hunk 2]
```diff
@@ -4,3 +4,86 @@
...
+
+Contains packaging scripts that generate the StimulusSets and assemblies 
+in BrainIO collection.
+
+The scripts herein are snapshots of code that added contents to brainio_collection.
```
Comment (mschrimpf) [inline README.md:13]: "The scripts in brainio_contrib"? (Or whatever that directory will be named)

File: brainio_collection/packaging.py [hunk 3]
```diff
@@ -67,6 +71,48 @@ def create_image_csv(proto_stimulus_set, target_path):
...
+
+def check_image_numbers(stimulus_set):
+    image_numbers = [int(image_file_path[image_file_path.rfind('_') + 1:image_file_path.rfind('.')])
+                     for image_file_path in list(stimulus_set.image_paths.values())]
+    image_numbers.sort()
+    for i in range(len(image_numbers) - 1):
+        assert image_numbers[i] == image_numbers[i + 1] - 1, "StimulusSet files not sequentially numbered"
+
+
+def check_stimulus_set(stimulus_set):
```
Comment (mschrimpf) [inline brainio_collection/packaging.py:99]: can we rename this to `check_experiment_stimulus_set` or so to make it clear that this is a special stimulus_set? Ideally also add a `"""` comment explaining what this does and what it's used for

**Review by mschrimpf — COMMENTED** [review]
* setup.py should exclude `brainio_contrib` from `find_packages` 
* maybe `notebooks` should go inside of `brainio_contrib`, check with @jjpr-mit. I guess ideally those would go inside the respective `brainio_contrib` packaging directories, but let's not get hung up with that
* restore transform files (https://github.com/brain-score/brainio_base/pull/17)

**Comment by stothe2** [comment]
Wait to merge until #55

**Review by mschrimpf — APPROVED** [review]

**Comment by mschrimpf** [comment]
After merging, we might want to use a fresh environment, `pip install brainio_collection` and see which files end up in `site-packages`. I believe the `setup.py` exclusion of the `brainio_contrib` folder should stop it from being copied but the [MANIFEST](https://github.com/brain-score/brainio_collection/blob/master/MANIFEST.in) might include it somehow

**Comment by mschrimpf** [comment]
closing and re-opening to re-trigger jenkins

**Pull request merged**